cpp/src/parquet/column_writer.cc

```diff
@@ -1468,42 +1468,43 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
   // which case we call back to the dense write path)
   std::shared_ptr<::arrow::Array> preserved_dictionary_;
 
-  int64_t WriteLevels(int64_t num_values, const int16_t* def_levels,
+  int64_t WriteLevels(int64_t num_levels, const int16_t* def_levels,
                       const int16_t* rep_levels) {
+    // Update histograms now, to maximize cache efficiency.
```

Comment:
Don't fully understand why scaning a pass of `def_levels` and `rep_levels` first would maximize cache efficiency, would `WriteDefinitionLevels` first also update the LiD cache?

Reply:
I'm not sure and I don't see any difference on our mini-benchmark. But `WriteDefinitionLevels` also writes to a destination buffer, so perhaps some of the levels will be evicted from the CPU cache.

Reply:
It seems like a description of why this is better would be helpful. Why is this more cache-friendly than what was there before?